dbms/src/Storages/Page/stress/workload/MainEntry.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <Storages/Page/stress/workload/PSStressEnv.h>
+#include <Storages/Page/stress/workload/PSWorkload.h>
+
+using namespace DB::PS::tests;
+
+int StressWorkload::mainEntry(int argc, char ** argv)
+{
+    {
+        // maybe due to sequence of linking, REGISTER_WORKLOAD is not visible to main function in dbms/src/Server/main.cpp
```

Comment:
why `__attribute__((constructor))` won't triggered before `mainEntry`?

Reply:
just as the comment says, maybe due to sequence of linking, REGISTER_WORKLOAD is not visible to main function in `dbms/src/Server/main.cpp`.
when we run in this way:
```shell
./dbms/src/Server/tiflash pageworkload -M 0 -I true -V 3 -P stress -T 120
```
` __attribute__((constructor))` won't be triggered.
we run in this way:
```shell
ninja page-workload
./dbms/src/Storages/Page/stress/workload/page-workload -M 0 -I true -V 3 -P stress -T 120
```
it will be triggered.
I still could not find the detail reason. 😢

Reply:
Maybe change the registration to be explicit like the function `registerStorages`? `__attribute__((constructor))` is not a common way and depends on specific compiler behavior.

https://github.com/pingcap/tiflash/blob/5b61ae70550624d3bf0b5ca6bac89013ed5a6a4b/dbms/src/Storages/registerStorages.cpp#L38-L56

Reply:
https://stackoverflow.com/questions/1202494/why-doesnt-attribute-constructor-work-in-a-static-library

Reply:
> Maybe change the registration to be explicit like the function `registerStorages`? `__attribute__((constructor))` is not a common way and depends on specific compiler behavior.

That's how we've settled it. explicitly trigger them.